plugins/list/plugin.js

```diff
@@ -1090,12 +1090,15 @@
 								// if we have two neighboring lists, merge the list items from the second
 								// into the first, rather than moving one item at a time.
 								var startItem = nextLine.startContainer;
+								var commonAncestor = li.getCommonAncestor(startItem);
 								if (
 									startItem.type == CKEDITOR.NODE_ELEMENT &&
 									startItem.getName() == 'li' &&
 									startItem.getParent().getName() in listNodeNames &&
-									!li.getParent().contains(startItem)
+									!( commonAncestor.getName() in listNodeNames ) &&
```

Comment:
superset of the previous case (if the list item we are currently in + the block being deleted into it have a common list ancesotr)